tests/tt_eager/python_api_testing/unit_testing/misc/test_split_any_dim_rm.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+import sys
+from loguru import logger
+import random
+import numpy as np
+import ttnn
+
+from models.utility_functions import (
+    comp_pcc,
+)
+import torch
+import sys
+import numpy
+import pytest
+import os
+import itertools
+
+debug = False
+
+# TODO: test other dims/num_splits
+two_chunk_dim_two_tests = list(
+    zip(
+        [[1, 2**k, 2] for k in range(10)],  # shapes
+        itertools.repeat(2),  # chunks
+        itertools.repeat(2),  # dim
+    )
+)
+
+two_chunk_dim_two_ids = [
+    "x".join(map(str, shape)) + f"@{dim}" + f"->{chunks}" for (shape, chunks, dim) in two_chunk_dim_two_tests
+]
+
+
+@pytest.mark.parametrize(
+    "in_mem_config",
+    (
+        ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM),
+        ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.L1),
+    ),
+    ids=["in_DRAM", "in_L1"],
+)
+@pytest.mark.parametrize(
+    "out_mem_config",
+    (
+        ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM),
+        ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.L1),
+    ),
+    ids=["out_DRAM", "out_L1"],
+)
+@pytest.mark.parametrize(
+    "refshape_chunks_dim",
+    tuple(two_chunk_dim_two_tests),
+    ids=two_chunk_dim_two_ids,
+)
+def test_split_rm(refshape_chunks_dim, in_mem_config, out_mem_config, device, dtype=ttnn.bfloat16):
+    (refshape, num_splits, dim) = refshape_chunks_dim
+    profile_location = "split_rm/"
+    os.system(f"rm -rf {profile_location}")
+
+    torch.manual_seed(1234)
+
+    Z = refshape[0]
+    Y = refshape[1]
+    X = refshape[2]
+
+    assert dim in [0, 1, 2]
+
+    if dim == 2:
+        chunk_shape = [Z, Y, X // num_splits]
+    elif dim == 1:
+        chunk_shape = [Z, Y // num_splits, X]
+    elif dim == 0:
+        chunk_shape = [Z // num_splits, Y, X]
+
+    logger.info(f"Split tensor of size: {str(refshape)}")
+
+    dtype_torch = torch.bfloat16
+
+    A = torch.arange(Z * Y * X, dtype=dtype_torch).reshape(refshape)
+    assert list(A.size()) == refshape
+
+    a_t = ttnn.from_torch(A, layout=ttnn.Layout.ROW_MAJOR, dtype=dtype, memory_config=in_mem_config, device=device)
```

Comment:
Can you clarify why [1, 256, 2] of bfloat16 works in Row Major?

Reply:
Multiples of 2 should work for bfloat16 for row major. Is this what you are referring to?

Reply:
the end goal though is splitting on the last dim which will be 1. which I think is why he is doing ROW-MAJOR, offline I thought though he was padding TILE tensors and handling it that way. This should work too though.

Reply:
Yeah, multiples of 2 work since the afaik the issue with small bfloat16 buffers is the requirement for 4 byte alignment on NOC transfers. 16 bits * 2 = 32 bits = 4 bytes. @ayerofieiev-tt. Of course, we do have a problem with row major bfloat16 buffers with inner dim 1; this is why the output of this op must be tiled when outputting on-device to match on-device input.

Reply:
@tarafdarTT I worked on the bilateral padding for tiled input version of this for a bit, but found that since the layout conversions have to happen on host there was no real performance gain in exchange for a lot of added complexity. This wound up being the cleaner change and moreover doesn't require the BERT team to change the format on their input tensor.